galaxy_importer/ansible_test/builders/local_image_build.py

```diff
@@ -0,0 +1,93 @@
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import logging
+import os
+import requests
+import subprocess
+import uuid
+
+from galaxy_importer import config
+from galaxy_importer import constants
+from galaxy_importer import exceptions
+from shutil import copy, copyfileobj, rmtree
+
+config_data = config.ConfigFile.load()
+cfg = config.Config(config_data=config_data)
+default_logger = logging.getLogger(__name__)
+
+
+class Build(object):
+    """Use docker/podman to build ansible-test image with artifact inside."""
+    def __init__(self, collection_file, collection_name, logger):
+        self.collection_file = collection_file
+        self.container_engine = Build.get_container_engine()
+        self.log = logger or default_logger
+        self.collection_name = collection_name
+        self.tag = f'galaxy-importer:{self.collection_name}'
+        self.working_dir = f'/tmp/galaxy-importer-{str(uuid.uuid4())}/'
+
+    @staticmethod
+    def get_container_engine():
+        if cfg.infra_local_docker is True:
+            return 'docker'
+        else:
+            return 'podman'
+
+    def build_dockerfile(self):
+        self.log.info('Building dockerfile')
+        url = 'https://raw.githubusercontent.com/ansible/galaxy-importer/master/docker/ansible-test/Dockerfile'
+        file_location = os.path.join(self.working_dir, 'Dockerfile')
+        with requests.get(url, allow_redirects=True) as r:
+            with open(file_location, 'wb') as f:
+                f.write(r.content)
+        with open(file_location, 'a') as f:
+            f.write('\nCOPY archive.tar.gz /archive/archive.tar.gz\n')
+
+    def build_image_with_artifact(self):
+        cmd = [self.container_engine, 'build', '.', '--quiet', '-t', self.tag]
+        try:
+            subprocess.run(cmd, cwd=self.working_dir)
+        except subprocess.CalledProcessError as e:
+            raise exceptions.AnsibleTestError(
+                'An exception occurred in: {}, message={}'.format(' '.join(cmd), e.msg))
+        return self.tag
+
+    def cleanup(self):
+        rmtree(self.working_dir)
+
+    def copy_collection_file(self):
+        path = os.path.join(self.working_dir, 'archive.tar.gz')
+
+        try:
+            url = Build._get_pulp_archive_url(self.pulp_artifact_file)
+            with requests.get(url, stream=True) as r:
+                with open(path, 'wb') as f:
+                    copyfileobj(r.raw, f)
+        except AttributeError:
+            spath = os.path.join(
+                os.path.dirname(constants.ROOT_DIR.rstrip('galaxy_importer')),
```

Comment:
this assumes the archive is in the path of the package, which won't be the case. the file passed in to the importer is accessible here as `self.collection_file`, but if we can use filepath noted above, that would be best.